src/Tokenizer/Tokens.php

```diff
@@ -133,17 +133,25 @@ public static function clearCache(?string $key = null): void
      */
     public static function detectBlockType(Token $token): ?array
     {
-        foreach (self::getBlockEdgeDefinitions() as $type => $definition) {
-            if ($token->equals($definition['start'])) {
-                return ['type' => $type, 'isStart' => true];
-            }
-
-            if ($token->equals($definition['end'])) {
-                return ['type' => $type, 'isStart' => false];
+        static $blockEdgeKinds = null;
+
+        if (null === $blockEdgeKinds) {
+            $blockEdgeKinds = [];
+            foreach (self::getBlockEdgeDefinitions() as $type => $definition) {
+                $blockEdgeKinds[
+                    \is_string($definition['start']) ? $definition['start'] : $definition['start'][0]
```

Comment:
Can you explain why the current code was not only wrapped with static variable, but also this kind of check was added?

Reply:
the PR changes the logic.

- master: loop over array till we find a match [O(n)]
- PR: convert array to Map, find the match via accessing exposed index directly [O(1)]

Reply:
I know it, I asked why there's support for `$definition['start'][0]` in this PR, because previously it was only checking `$definition['start']`. But OK, I overlooked `getBlockEdgeDefinitions()`'s return type, so it's clear now.

Reply:
ah, I didn't understood you were asking for `[0]` in initial msg. 

good it's clear now!